holoviews/operation/datashader.py

```diff
@@ -1789,12 +1789,7 @@ def _process(self, element, key=None):
 
             for k, v in sel_data.items():
                 new_data[k].data = v
-
-            # TODO: Investigate why this does not work
-            # element = element.clone(data=new_data, kdims=element.vdims.copy(), vdims=element.vdims.copy())
```

Comment:
Fresh eyes showed that `kdims=element.vdims.copy()` was the culprit; it should have been kdims.

Reply:
So there's no need to pass kdims and vdims to `clone()` (I assume so, just checking).

Reply:
There shouldn't be. Have tested with the example used in  #6422 and saw no error.